oshi-core/src/main/java/oshi/software/os/linux/LinuxOSProcess.java

```diff
@@ -312,25 +322,11 @@ public boolean updateAttributes() {
         this.groupID = ParseUtil.whitespaces.split(status.getOrDefault("Gid", ""))[0];
         this.group = UserGroupInfo.getGroupName(groupID);
         this.name = status.getOrDefault("Name", "");
-        switch (status.getOrDefault("State", "U").charAt(0)) {
-        case 'R':
-            this.state = State.RUNNING;
-            break;
-        case 'S':
-            this.state = State.SLEEPING;
-            break;
-        case 'D':
-            this.state = State.WAITING;
-            break;
-        case 'Z':
-            this.state = State.ZOMBIE;
-            break;
-        case 'T':
-            this.state = State.STOPPED;
-            break;
-        default:
-            this.state = State.OTHER;
-            break;
+        this.state = ProcessStat.getState(status.getOrDefault("State", "U").charAt(0));
+        List<Integer> threadIds = ProcessStat.getThreadIds(getProcessID());
+        this.threadDetails = new ArrayList<OSThread>();
+        for (Integer threadId : threadIds) {
+            this.threadDetails.add(new LinuxOSThread(this, threadId));
```

Comment:
I don't think you need to assign the `threadIDs` temporary variable, just include it in the foreach clause.  Although I think these 3 lines would read more clearly as a stream.  Something like this (untested):

this.threadDetails = ProcessStat.getThreadIds(getProcessID()).stream().map(id -> new LinuxOSThread(this, id)).collect(Collectors.toList()));

Reply:
Thats a nice way to do :) Will do it.